frontend/packages/kubevirt-plugin/src/selectors/event/filters.ts

```diff
@@ -9,64 +8,55 @@ import {
-
-  return (event) =>
-    _.isMatch(event, {
-      name: vmName,
-      namespace: vmNamespace,
-      kind: VirtualMachineInstanceModel.kind,
-    });
-};
+const vmiEventFilter = (vm: VMKind | VMIKind) => ({ kind, namespace, name }: EventFilterProps) =>
+  kind === VirtualMachineInstanceModel.kind &&
+  name === getName(vm) &&
+  namespace === getNamespace(vm);
 
-export const launcherPodEventFilter = (vm: VMKind) => {
+const launcherPodEventFilter = (vm: VMKind) => {
   const podNameStart = `${VIRT_LAUNCHER_POD_PREFIX}${getName(vm)}-`;
-  const vmNamespace = getNamespace(vm);
 
-  return ({ kind, namespace, name }) =>
-    kind === PodModel.kind && namespace === vmNamespace && name.startsWith(podNameStart);
+  return ({ kind, namespace, name }: EventFilterProps) =>
+    kind === PodModel.kind && namespace === getNamespace(vm) && name.startsWith(podNameStart);
 };
 
-export const importerPodEventFilter = (vm: VMKind) => {
-  const vmName = getName(vm);
-  const vmNamespace = getNamespace(vm);
-
-  return ({ kind, namespace, name }) => {
-    // importer pod example importer-<diskName>-<vmname>-<generatedId>
-    // note: diskName and vmname may contain '-' which means pod name should have at least 4 parts
-    if (
-      kind === PodModel.kind &&
-      namespace === vmNamespace &&
-      name.startsWith('importer-') &&
-      name.split('-').length > 3
-    ) {
-      const importerDashIndex = name.indexOf('-');
-      const diskDashIndex = name.indexOf('-', importerDashIndex + 1);
-      const lastDashIndex = name.lastIndexOf('-');
-      // try to remove importer- and some part of <diskname>
-      const diskAndVmName = name.slice(diskDashIndex + 1, lastDashIndex);
-      return diskAndVmName.endsWith(vmName);
-    }
-    return false;
-  };
+const importerPodEventFilter = (vm: VMKind) => ({ kind, namespace, name }: EventFilterProps) => {
+  // importer pod example importer-<diskName>-<vmname>-<generatedId>
+  // note: diskName and vmname may contain '-' which means pod name should have at least 4 parts
+  if (
+    kind === PodModel.kind &&
+    namespace === getNamespace(vm) &&
+    name.startsWith('importer-') &&
+    name.split('-').length > 3
+  ) {
+    const importerDashIndex = name.indexOf('-');
+    const diskDashIndex = name.indexOf('-', importerDashIndex + 1);
+    const lastDashIndex = name.lastIndexOf('-');
+    // try to remove importer- and some part of <diskname>
+    const diskAndVmName = name.slice(diskDashIndex + 1, lastDashIndex);
+    return diskAndVmName.endsWith(getName(vm));
+  }
+  return false;
 };
 
-export const vmiMigrationEventFilter = (vm: VMKind) => {
-  const vmName = getName(vm);
-  const vmNamespace = getNamespace(vm);
-
-  return ({ kind, namespace, name }) =>
-    kind === VirtualMachineInstanceMigrationModel.kind &&
-    namespace === vmNamespace &&
-    name === `${vmName}-migration`;
+const vmiMigrationEventFilter = (vm: VMKind) => ({ kind, namespace, name }: EventFilterProps) =>
```

Comment:
wouldnt make sense to reuse `EventInvolvedObject` here ?
```suggestion
const vmiMigrationEventFilter = (vm: VMKind) => ({ kind, namespace, name }: EventInvolvedObject) =>
```

Reply:
After second look, return types for filter functions are added. The `EventInvolvedObject` is used as well.